CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+
+Alternatively, read some of the [many docs on the system](https://tyk.io/tyk-documentation/get-started/), and pick a component that seems
+interesting.  Start with `main()` and read
+until you find something you want to fix.  The best way to learn is to hack!
+There's always code that can be clarified and variables or functions that can
+be renamed or commented.
+
+### `master` is unsafe
+
+We will do our best to keep master in good shape, with tests passing at all times. But in order to move fast, we will make API changes that your application might not be compatible with. We will do our best to communicate these changes and version appropriately so you can lock into a specific version if need be. For stable releases check our tags and `stable` branch, which contains our latest stable release.
+
+
+### Contributing A Patch
+
+If you're working on an existing issue, such as one of the `help-wanted` ones
+above, simply respond to the issue and express interest in working on it.  This
+helps other people know that the issue is active, and hopefully prevents
+duplicated efforts.
+
+If you want to work on a new idea of relatively small scope:
+
+1. Submit an issue describing your proposed change to the repo in question.
+1. The repo owners will respond to your issue promptly.
+1. Clone the repo, develop, and test your changes.
+1. Submit a pull request.
+1. If your proposed change is accepted, and you haven't already done so, sign a
+   Contributor License Agreement (see details above).
+
+If you want to work on a bigger idea, we STRONGLY recommend that you start with
+some bugs or smaller features. It is always better to discuss your idea with our team first, before implementing it.
+
+### Downloading the project
+You need to clone Tyk from GitHub to your GOPATH folder, or alternatively you can run `go get -d github.com/TykTechnologies/tyk` which automatically downloads project to the right path.
+
+### Building the project
+You need to have working Go enironment: see [golang.org](https://golang.org/doc/code.html) for more info on how Go works with code.
+
+To build and test Tyk use built-in `go` commands: `go build` and `go test -v`. If you want to just test a subset of the project, you can pass the `-run` argument with name of the test.
+
+Currently in order for tests to pass, a **Redis host is required**. We know, this is terrible and should be handled with an interface, and it is, however in the current version there is a hard requirement for the application to have its default memory setup to use redis as part of a deployment, this is to make it easier to install the application for the end-user. Future versions will work around this, or we may drop the memory requirement. Simplest way to run Redis is to use official Docker image [https://hub.docker.com/_/redis/](https://hub.docker.com/_/redis/)
+
+### Adding dependencies
+
+If your patch depends on new packages, ensure that they will be put in `/vendor` folder.
```

Comment:
If we're to add docs about vendoring, we should at least mention what tool we use. Otherwise I would remove this section.